src/web/common.rkt

```diff
@@ -67,44 +67,135 @@
                               ,@(for/list ([line extra])
                                   `(li ,line)))))))))
 
+(define (render-comparison #:title [title #f] name a b )
+  (render-large #:title title name a `(span ([class "unit"]) " → ") b))
+
 (define (render-large #:title [title #f] name . values)
   `(div ,name ": " (span ([class "number"]
                           ,@(if title `([title ,title]) '()))
                          ,@values)))
 
-(define (render-comparison #:title [title #f] name a b )
-  (render-large #:title title name a `(span ([class "unit"]) " → ") b))
+(define (preprocess-sort-core output sort-preprocesses)
+  (string-append
+   ";; Ensure these are sorted, for example in Racket, do\n"
+   (apply
+    string-append
+    (for/list ([variables sort-preprocesses])
+      (format "~a\n" `(match-define (list ,@variables) (sort ,variables <)))))
+   "\n"
+   output))
+
+(define (preprocess-sort-c-like output sort-preprocesses #:assert-has-parentheses [assert-has-parentheses #t])
+  ;; Stolen from https://github.com/codereport/racket-algorithms/blob/master/main.rkt#L88-L96
+  (define (sliding lst size [step 1])
+    (define (tail-call lst)
+      (if (>= size (length lst))
+          (list lst)
+          (cons (take lst size)
+                (tail-call (drop lst step)))))
+    (if (>= step (length lst))
+        (error "step has to be smaller then length of the list")
+        (tail-call lst)))
```

Comment:
This is wildly overgeneral. Just do `(for/list ([v1 (in-list variables)] [v2 (in-list (cdr variables)]) ...)`.

Reply:
Ah much cleaner, thank you